orders-core/src/main/java/io/helidon/examples/sockshop/orders/OrderResource.java

```diff
@@ -100,10 +103,35 @@ public Response newOrder(@Context UriInfo uriInfo, NewOrderRequest request) {
             throw new InvalidOrderException("Invalid order request. Order requires customer, address, card and items.");
         }
 
-        List<Item> items    = httpGet(request.items, new GenericType<List<Item>>() { });
-        Address    address  = httpGet(request.address, Address.class);
-        Card       card     = httpGet(request.card, Card.class);
-        Customer   customer = httpGet(request.customer, Customer.class);
+        String itemsId = request.items;
+        String cardId = request.card;
+        String addressId = request.address;
+        String customerId = request.customer;
+
+        /*
+         * Workaround for the time being - until the respective callers pass the data, not the full URL
+         */
+        int idx = itemsId.indexOf("/carts/");
+        if (idx >= 0 && itemsId.endsWith("/items")) {
+            itemsId = itemsId.substring(idx + 7, itemsId.length() - 6);
+        }
+        idx = cardId.indexOf("/cards/");
+        if (idx >= 0) {
+            cardId = cardId.substring(idx + 7);
+        }
+        idx = addressId.indexOf("/addresses/");
+        if (idx >= 0) {
+            addressId = addressId.substring(idx + 11);
+        }
+        idx = customerId.indexOf("/customers/");
+        if (idx >= 0) {
+            customerId = customerId.substring(idx + 11);
+        }
```

Comment:
It seems adding accessors to `NewOrderRequest` interact with other things (I've encountered `toString` - probably autogenerated by `@ToString`, and some more obscure issue that surfaced in `orders-redis` specifically - something to do with JSON serialization or deserialization). Because adding them is a disruptive change, I propose to address this as a separate issue.

Reply:
For posterity:

when exposing `getItems` directly, `toString` crashes - it uses these accessor methods instead of reading the fields directly;

when trying to expose accessors that are not exactly the names of the fields, eg `getItemsId` instead of `getItems`:
```[INFO] Running io.helidon.examples.sockshop.orders.redis.RedisOrderResourceIT
[ERROR] Tests run: 7, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 10.627 s <<< FAILURE! - in io.helidon.examples.sockshop.orders.redis.RedisOrderResourceIT
[ERROR] testInvalidOrder  Time elapsed: 0.027 s  <<< ERROR!
com.fasterxml.jackson.databind.JsonMappingException: (was java.lang.NullPointerException) (through reference chain: io.helidon.examples.sockshop.orders.NewOrderRequest["itemsId"])
Caused by: java.lang.NullPointerException```

Reply:
Sure. Let's commit this as-is then and I'll clean it up and deal with the fallout.